### PR TITLE
fix: creating config file

### LINKED ISSUE #1017: Installing testkube does not create .testkube/config.json
**Describe the bug**
When installing/updating testkube, the following error appears from not being able to read the Testkube config file. 
```
{
  "level": "error",
  "ts": 1645557105.601299,
  "caller": "config/config.go:20",
  "msg": "can't load configuration file",
  "error": "open /Users/abdallahabedraba/.testkube/config.json: no such file or directory",
  "stacktrace": "github.com/kubeshop/testkube/cmd/kubectl-testkube/config.init.0\n\t/home/runner/work/testkube/testkube/cmd/kubectl-testkube/config/config.go:20\nruntime.doInit\n\t/opt/hostedtoolcache/go/1.17.7/x64/src/runtime/proc.go:6498\nruntime.doInit\n\t/opt/hostedtoolcache/go/1.17.7/x64/src/runtime/proc.go:6475\nruntime.doInit\n\t/opt/hostedtoolcache/go/1.17.7/x64/src/runtime/proc.go:6475\nruntime.doInit\n\t/opt/hostedtoolcache/go/1.17.7/x64/src/runtime/proc.go:6475\nruntime.main\n\t/opt/hostedtoolcache/go/1.17.7/x64/src/runtime/proc.go:238"
}
```

Have mitigated for now by creating a `config.json` with an empty object `{}`

**To Reproduce**
1. Delete current testkube and config file
2. Install testkube again
3. Run `testkube version`

**Expected behavior**
Testkube correctly setup

**Version / Cluster**
- Which testkube version? 0.9.11
- OS: MacOS

**Screenshots**
image
 
**Additional context**

### PR DESCRIPTION
**Pull request by exu**
## Pull request description 

closes #1017 

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: cmd/kubectl-testkube/config/config.go [hunk 1]
```diff
@@ -10,7 +10,11 @@ var Config config
 func init() {
 	Config = config{}
 	l := log.DefaultLogger
-	err := Config.Init()
+
+	// set default analytics enabled
+	Config.Data.AnalyticsEnabled = true
+
+	err := Config.Init(Config.Data)
```
Comment (vsukhin) [inline cmd/kubectl-testkube/config/config.go:17]: It's a bit weird, Config methods have access to Data field, it's even public. Init and Save should have a parameter, if we plan to pass non Data field

File: cmd/kubectl-testkube/config/wtiter.go [hunk 2]
```diff
@@ -27,26 +27,41 @@ func (c *Storage) Save(data Data) error {
...
-func (c *Storage) Init() error {
+func (c *Storage) Init(data Data) error {
 	// create ConfigWriter dir if not exists
-	if dir := c.getDir(); dir != "" {
-		_, err := os.Stat(dir)
-		if os.IsNotExist(err) {
-			return os.Mkdir(dir, 0700)
+	dir := c.getDir()
+	_, err := os.Stat(dir)
+	if os.IsNotExist(err) {
```
Comment (vsukhin) [inline cmd/kubectl-testkube/config/wtiter.go:37]: if it's other error, then return

File: cmd/kubectl-testkube/config/wtiter.go [hunk 3]
```diff
@@ -27,26 +27,41 @@ func (c *Storage) Save(data Data) error {
...
 }
 
 func (c *Storage) getDir() string {
-	dirname, err := os.UserHomeDir()
-	if err != nil {
-		return ""
-	}
-
-	return path.Join(dirname, dir)
+	home, _ := os.UserHomeDir()
```
Comment (vsukhin) [inline cmd/kubectl-testkube/config/wtiter.go:61]: I prefer always check errors, even for simplest methods

File: cmd/kubectl-testkube/config/wtiter.go [hunk 4]
```diff
@@ -27,26 +27,41 @@ func (c *Storage) Save(data Data) error {
...
+		err := os.Mkdir(dir, 0700)
+		if err != nil {
+			return err
 		}
 	}
+
+	// create empty JSON file if not exists
+	path := c.getPath()
+	_, err = os.Stat(path)
+	if os.IsNotExist(err) {
```
Comment (vsukhin) [inline cmd/kubectl-testkube/config/wtiter.go:47]: if it's other error, then return

File: cmd/kubectl-testkube/config/wtiter.go [hunk 5]
```diff
@@ -27,26 +27,41 @@ func (c *Storage) Save(data Data) error {
...
+		f, err := os.Create(path)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
```
Comment (vsukhin) [inline cmd/kubectl-testkube/config/wtiter.go:52]: you don't need defer there, just close file before calling Save

**Review by vsukhin — COMMENTED** [review]
a few comments

**Comment by exu** [comment]
also removed this strange composition - I think I was thinking about something else - and it looked like incomplete code ;)

**Review by vsukhin — APPROVED** [review]

**Pull request merged**